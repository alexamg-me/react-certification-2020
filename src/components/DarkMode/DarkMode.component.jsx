import React, { useContext } from 'react';
import { Wrapper, Switch, Track, Ball } from './DarkMode.styled';
import { GlobalContext } from '../../context/GlobalContext';

function DarkMode({ children }) {
  const globalContext = useContext(GlobalContext);

  const onClickHandler = () => {
    globalContext.toggleTheme();
  };
  return (
    <Wrapper>
      <Switch onClick={onClickHandler}>
        <Track theme={globalContext.colors} />
        <Ball className={globalContext.themeId} />
      </Switch>
      {children}
    </Wrapper>
  );
}

export default DarkMode;
