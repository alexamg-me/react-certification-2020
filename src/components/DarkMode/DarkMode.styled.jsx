import styled from 'styled-components';

const Wrapper = styled.div`
  display: inline-block;
  padding: 0 20px;
  height: 40px;
`;
const Switch = styled.div`
  display: inline-block;
  position: relative;
  margin: 0 20px;
`;
const Track = styled.div`
  display: block;
  width: 50px;
  height: 25px;
  border-radius: 100px;
  background-color: ${(props) => props.theme.background || 'red'};
  transition: all 0.4s ease-in-out;
`;
const Ball = styled.div`
  display: block;
  width: 20px;
  height: 20px;
  background-color: #409287;
  border-radius: 100px;
  position: absolute;
  z-index: 1;
  top: calc(50% - 10px);
  left: 5px;
  transition: all 0.4s ease-in-out;
  &.light {
    background-color: #222222;
  }
  &.dark {
    background-color: #dddddd;
    left: calc(100% - 25px);
  }
`;

export { Wrapper };
export { Switch };
export { Track };
export { Ball };
