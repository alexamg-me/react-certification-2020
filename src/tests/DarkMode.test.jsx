import React from 'react';
import TestRenderer from 'react-test-renderer';
import DarkMode from '../components/DarkMode/DarkMode.component';

describe(`Component: DarkMode`, () => {
  test(`DarkMode renders with default props`, () => {
    const component = TestRenderer.create(<DarkMode />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
