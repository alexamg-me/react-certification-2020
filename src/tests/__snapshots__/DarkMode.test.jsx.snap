// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DarkMode DarkMode renders with default props 1`] = `
<div
  className="sc-bdfBwQ goSDW"
>
  <div
    className="sc-gsTCUz dIgXFX"
  >
    <div
      className="sc-dlfnbm tFEro"
    />
    <div
      className="sc-hKgILt dxvrxJ"
    />
  </div>
  <div
    className="sc-eCssSg bKUxkL"
  >
    Dark Mode
  </div>
</div>
`;
